src/GitVersionTask.MsBuild/LibGit2Sharp/GitLoaderContext.cs

```diff
@@ -11,22 +10,28 @@ namespace GitVersion.MSBuildTask.LibGit2Sharp
 {
     public sealed class GitLoaderContext : AssemblyLoadContext
     {
-        private readonly string[] assemblies;
+        private readonly Assembly entryPointAssembly;
         public static GitLoaderContext Instance { get; private set; }
 
-        private GitLoaderContext(string[] assemblies) => this.assemblies = assemblies;
+        private GitLoaderContext(Assembly entryPointAssembly) => this.entryPointAssembly = entryPointAssembly;
 
-        public static void Init(params string[] assemblies) => Instance = new GitLoaderContext(assemblies);
+        public static void Init(Assembly entryPointAssembly) => Instance = new GitLoaderContext(entryPointAssembly);
 
         protected override Assembly Load(AssemblyName assemblyName)
         {
-            if (assemblies.Contains(assemblyName.Name))
+            if (assemblyName.Name == entryPointAssembly.GetName().Name)
```

Comment:
```suggestion
            if (assemblyName.Name.Equals(entryPointAssembly.GetName().Name))
```
Also, shouldn't we take into account OS to know if we can be case insensitive?